src/main/java/com/atlassian/bitbucket/jenkins/internal/applink/oauth/serviceprovider/rest/TokenCrumbExclusion.java

```diff
@@ -0,0 +1,29 @@
+package com.atlassian.bitbucket.jenkins.internal.applink.oauth.serviceprovider.rest;
+
+import hudson.Extension;
+import hudson.security.csrf.CrumbExclusion;
+
+import javax.servlet.FilterChain;
+import javax.servlet.ServletException;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import java.io.IOException;
+
+import static org.apache.commons.lang3.StringUtils.isEmpty;
+
+@Extension
+public class TokenCrumbExclusion extends CrumbExclusion {
```

Comment:
We already have a `com.atlassian.bitbucket.jenkins.internal.applink.oauth.serviceprovider.rest.OauthCrumbExclusion` maybe we could add this check to it? It seems like this is just an additional case that the `OauthCrumbExclusion` should have handled?

Reply:
I was basing this off of the design for the webhook exception, and thought it was "one exception per case". That said, they are very similar- happy to migrate the code into that class to keep things simple :)

Reply:
I think the resources are per "task" so anything for Oauth should be in the oauth filter.

Reply:
@mhenschke-atl we don't need this file any more, right? nor the test file for it.

Reply:
We don't :facepalm: I didn't push the delete. Sorry!